dissolve-struct-lib/src/main/scala/ch/ethz/dalab/dissolve/classification/StructSVMModel.scala

```diff
@@ -31,11 +31,42 @@ class StructSVMModel[X, Y](
     dissolveFunctions: DissolveFunctions[X, Y]) =
     this(weights, ell, ellMat, dissolveFunctions, -1)
 
+  def addToWeight(delta:Vector[Double]): Unit = {
+    //if delta is sparse we can speedup the addition even more
+     delta match{
+       case delta:SparseVector[Double] => 
+         val indexes = delta.index
+         for(i <- 0 until indexes.length){
+           val idx = indexes(i)
+           weights(idx) += delta.data(i)
+         }
+       case _ => 
+         for(i <- 0 until delta.length){
+           weights(i) += delta(i)
+         }
+     }
+  }
+  
+  def subtractFromWeight(delta:Vector[Double]): Unit = {
+     delta match{
+       case delta:SparseVector[Double] => 
+         val indexes = delta.index
+         for(i <- 0 until indexes.length){
+           val idx = indexes(i)
+           weights(idx) -= delta.data(i)
+         }
+       case _ => 
+         for(i <- 0 until delta.length){
+           weights(i) -= delta(i)
+         }
+     }
+  }
+    
```

Comment:
would be good to just give the breeze += operator a try that one has the same task. then only if there are notable differences in GC or performance, it makes sense to implement your own (in a separate helper class)